src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -123,74 +124,85 @@ public WebHookReceiverManager GetWebHookReceiverManager(WebHostSettings settings
         /// </summary>
         internal void EnsureInitialized(WebHostSettings settings)
         {
-            lock (_syncLock)
+            ILoggerFactory defaultLoggerFactory = GetDefaultLoggerFactory(settings);
+            _logger = defaultLoggerFactory.CreateLogger(ScriptConstants.LogCategoryHostStartup);
```

Comment:
Create the logger in constructor?

Reply:
Creating GetDefaultLoggerFactory depends on 'WebHostSettings settings' which is not a DI dependency for `WebHostResolver`
@fabiocav - is it ok to update dependencies for `WebHostResolver` ?